R/Renderer.R

```diff
@@ -33,10 +33,16 @@ Renderer <- R6::R6Class( # nolint: object_name_linter.
       if (missing(yaml_header)) {
         yaml_header <- md_header(yaml::as.yaml(list(title = "Report")))
       }
-      parsed_yaml <- yaml_header
+
+      private$report_type <- reverse_yaml_field(yaml_header, "output")
       parsed_global_knitr <- sprintf(
-        "\n```{r setup, include=FALSE}\nknitr::opts_chunk$set(%s)\n```\n",
-        capture.output(dput(global_knitr))
+        "\n```{r setup, include=FALSE}\nknitr::opts_chunk$set(%s)\n%s\n```\n",
+        capture.output(dput(global_knitr)),
+        if (identical(private$report_type, "powerpoint_presentation")) {
+          paste(capture.output(code_block_function), collapse = "\n")
```

Comment:
This captures output of the function definition and not a function call? :thinking:

Reply:
It can captures the definition, but I was also obtaining bytecode and environment along with it , so I am using deparse  instead.